tutorial-server/notebooks/CactusTutorial.ipynb

```diff
@@ -135,7 +135,7 @@
    "outputs": [],
    "source": [
     "%%bash\n",
-    "[ -r ~etuser/Cactus.tar.gz ] && mkdir ~/Cactus && tar -xzf ~etuser/Cactus.tar.gz -C ~/"
+    "[ -r ~etuser/Cactus.tar.gz ] && mkdir ~/Cactus/repos && tar -xzf ~etuser/Cactus.tar.gz -C ~/"
```

Comment:
I think this should be `mkdir -p ~/Cactus/repos`

Reply:
Uh, tricky. I don't want `-p` since it does not return an error if the directory exists and I abuse `mkdir` failure to detect the existence of a Cactus tree. Now at this point I actually already have a `~/Cactus` directory since I create one in the notebook preamble (so that `PYTHONUSERHOME` gets used). Highly annoying.   
`~/Cactus` is created in 03a1df3 line 106 (note the use of `makedirs` vs. `makedir`):

```python
os.makedirs(site.USER_SITE, exist_ok=True)
```

This is also why I cannot use `mkdir ~/Cactus`.

_However_ thinking about this... I don't need `mkdir` I should just use a `! [ -d ~/Cactus/repos ]` since this is really what I want (more characters though ;-) ).

I'll update the pull request. Thanks for the suggestion.

Reply:
Replaced by

```shell
[ -r ~etuser/Cactus.tar.gz ] && ! [ -d ~/Cactus/repos ] && tar -xzf ~etuser/Cactus.tar.gz -C ~/"
```